docs/network-binding-plugin.md

```diff
@@ -0,0 +1,342 @@
+    }
+
+    updatedDomainSpec, err := changeDomain(domainSpec, vmi)
+    if err != nil {
+        return nil, err
+    }
+
+    newDomainXML, err := xml.Marshal(updatedDomainSpec)
+    if err != nil {
+        return nil, fmt.Errorf("failed to marshal updated domain spec: %v", err)
+    }
+
+    return &hooksv1alpha2.OnDefineDomainResult{
+        DomainXML: newDomainXML,
+    }, nil
+}
+```
+
+It is recommended to use the latest available hook version.
+The unused methods must be defined to satisfy the interface, preferably with
+a default non-error response.
+
+#### Pod interface name
+
+The name of the network interface in the pod, to which the relevant network
+is to bind to, is expected to be known.
+
+- For the primary network, the default `eth0` naming is expected.
+- For the secondary networks, Kubevirt explicitly specifies the name through
+  Multus annotations of the pod. By that it assumes the CNI plugin used
+  confirms with the CNI specification and will name the interface accordingly.
+  The name used is generated by Kubevirt with the following format:
+  `pod<network name hash ID>`.
+  - Network hash ID: The hash ID is generated based on the network name specified
+    in the VM/VMI interface & network spec.
+    To programmatically generate it, one can use the following code snippet:
+
+```go
+import "kubevirt.io/kubevirt/pkg/network/namescheme"
+
+ifaceName := namescheme.GenerateHashedInterfaceName(network_name)
+```
+
+#### Domain configuration
+
+Changing the domain configuration usually involves the addition of a device
+or a change in an existing device.
+
+Either way, it is important to understand which are the networking
+devices and what are the expectations from Kubevirt:
+
+- Domain API: "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+  - `DomainSpec`: The domain configuration spec structure, which is in sync
+    with [libvirt domain configuration](https://libvirt.org/formatdomain.html).
+    - `Devices`:
+      - `Interface`: The network interface configuration spec.
+      - `HostDevice`: The general host device configuration spec.
+        The host-device is used to pass-through devices from the host to the domain.
+        SR-IOV virtual functions are a classic example for networking.
+- Device Alias:
+  Kubevirt marks on the domain device spec an alias that references
+  the logical network name as appearing in the VM/VMI spec.
+  - For SR-IOV devices, the alias also includes a prefix: `sriov-`.
+
+  ```go
+  import domainschema "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+  deviceAlias := domainschema.NewUserDefinedAlias("sriov-" + "mynet1")
+  ```
+- Parameters: Several configuration parameters of an interface from the VMI spec
+  are passed to the domain device configuration directly (e.g. the mac address).
+  The domain interface and host-device expose a larger amount of parameters
+  which are not exposed through the VMI interface spec.
```

Comment:
```suggestion
  The domain interface and host-device may expose additional  parameters
  which are not exposed through the VMI interface spec.

Reply:
`may` is not a fit, but I did drop the "larger amount".

Done